pkg/reconciler/awssnssource/subscription.go

```diff
@@ -127,39 +143,48 @@ func (r *Reconciler) ensureUnsubscribed(ctx context.Context) error {
 	case isNotFound(err):
 		// the finalizer is unlikely to recover from a missing Secret,
 		// so we simply record a warning event and return
-		event.Warn(ctx, ReasonFailedUnsubscribe, "Secret missing while finalizing subscription %q. Ignoring: %s",
-			*subsARN, err)
+		event.Warn(ctx, ReasonFailedUnsubscribe,
+			"Secret missing while finalizing event source. Ignoring: %s", err)
 		return nil
 	case err != nil:
 		return fmt.Errorf("%w", reconciler.NewEvent(corev1.EventTypeWarning, ReasonFailedUnsubscribe,
 			"Error creating SNS client: %s", err))
 	}
 
-	resp, err := snsClient.UnsubscribeWithContext(ctx, &sns.UnsubscribeInput{
-		SubscriptionArn: subsARN,
-	})
+	topicARN := spec.ARN.String()
 
+	subsARN, err := findSubscription(ctx, snsClient, topicARN, url.String())
 	switch {
 	case isNotFound(err):
 		return reconciler.NewEvent(corev1.EventTypeNormal, ReasonUnsubscribed,
-			"Subscription %q already absent, skipping finalization", *subsARN)
```

Comment:
We do the same in the finalizer, and return early if no subscription exists for our criteria.